WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderBlogInfoView.java

```diff
@@ -0,0 +1,181 @@
+                }
+            };
+            if (mFeedId != 0) {
+                ReaderBlogActions.updateFeedInfo(mFeedId, null, listener);
+            } else {
+                ReaderBlogActions.updateBlogInfo(mBlogId, null, listener);
+            }
+        }
+    }
+
+    private void showBlogInfo(ReaderBlog blogInfo) {
+        // do nothing if unchanged
+        if (blogInfo == null || blogInfo.isSameAs(mBlogInfo)) {
+            return;
+        }
+
+        mBlogInfo = blogInfo;
+
+        ViewGroup layoutInfo = (ViewGroup) findViewById(R.id.layout_blog_info);
+        ViewGroup layoutDescription = (ViewGroup) layoutInfo.findViewById(R.id.layout_blog_description);
+        TextView txtBlogName = (TextView) layoutInfo.findViewById(R.id.text_blog_name);
+        TextView txtDomain = (TextView) layoutInfo.findViewById(R.id.text_domain);
+        TextView txtDescription = (TextView) layoutInfo.findViewById(R.id.text_blog_description);
+        TextView txtFollowCount = (TextView) layoutInfo.findViewById(R.id.text_blog_follow_count);
+
+        if (blogInfo.hasName()) {
+            txtBlogName.setText(blogInfo.getName());
+        } else {
+            txtBlogName.setText(R.string.reader_untitled_post);
+        }
+
+        if (blogInfo.hasUrl()) {
+            txtDomain.setText(UrlUtils.getDomainFromUrl(blogInfo.getUrl()));
+            txtDomain.setVisibility(View.VISIBLE);
+        } else {
+            txtDomain.setVisibility(View.GONE);
+        }
+
+        if (blogInfo.hasDescription()) {
+            txtDescription.setText(blogInfo.getDescription());
+            layoutDescription.setVisibility(View.VISIBLE);
+        } else {
+            layoutDescription.setVisibility(View.GONE);
+        }
+
+        txtFollowCount.setText(String.format(getContext().getString(R.string.reader_label_follow_count), blogInfo.numSubscribers));
+
+        mFollowButton.setIsFollowed(blogInfo.isFollowing);
+        mFollowButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                toggleFollowStatus();
+            }
+        });
+
+        if (layoutInfo.getVisibility() != View.VISIBLE) {
+            layoutInfo.setVisibility(View.VISIBLE);
+        }
+
+        if (mBlogInfoListener != null) {
+            mBlogInfoListener.onBlogInfoLoaded(blogInfo);
+        }
+    }
+
+    private void toggleFollowStatus() {
+        if (!NetworkUtils.checkConnection(getContext())) {
+            return;
+        }
+
+        final boolean isAskingToFollow;
+        if (mFeedId != 0) {
+            isAskingToFollow = !ReaderBlogTable.isFollowedFeed(mFeedId);
+        } else {
+            isAskingToFollow = !ReaderBlogTable.isFollowedBlog(mBlogId);
+        }
+
+        ReaderActions.ActionListener listener = new ReaderActions.ActionListener() {
+            @Override
+            public void onActionResult(boolean succeeded) {
+                if (!succeeded && getContext() != null) {
```

Comment:
Maybe we should hold a reference to the listener and limit the requests to one at a time. Pressing it twice in succession can lead to contradictory toast messages. I'm assuming the same could happen on a slow connection.